jenkins/spark-tests.sh

```diff
@@ -273,7 +275,7 @@ run_pyarrow_tests() {
 
 run_non_utc_time_zone_tests() {
   # select one time zone according to current day of week
-  source "$(dirname "$0")"/test-timezones.sh
+  source "${WORKSPACE}/jenkins/test-timezones.sh"
```

Comment:
$0 is the relative path, when we run the test script `jenkins/spark_test.sh`, $0 is `jenkins/` but currently $PWD path is `integration-tests/jars/` with pushd: https://github.com/NVIDIA/spark-rapids/blob/branch-24.02/jenkins/spark-tests.sh#L137

Reply:
change to use absolute path $WORKSPACE/jenkins/test-timezones.sh